### PR TITLE
Upgrade go version to 1.22.

### PR DESCRIPTION
**Pull request by xuezhaojun**

### CODE REVIEW

**Comment by haoqing0110** [comment]
I've encountered the problem that the binary built with golang 1.22 can not run on ubi8 and need to upgrade the base image to ubi9. Feel free to merge this PR if it does not have such issue.

**Comment by haoqing0110** [comment]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,27 +1,28 @@
 module github.com/stolostron/cluster-proxy-addon
 
-go 1.21
+go 1.22.0
```
Comment (haoqing0110) [inline go.mod:3]: cicd supports up to go 1.22.5.

Comment (xuezhaojun) [reply go.mod]: Updated.

**Comment by xuezhaojun** [comment]
> I've encountered the problem that the binary built with golang 1.22 can not run on ubi8 and need to upgrade the base image to ubi9. Feel free to merge this PR if it does not have such issue.

Great point, thanks! I updated the image to ubi9!

**Comment by xuezhaojun** [comment]
@haoqing0110 PR updated, please take another look.

**Comment by haoqing0110** [comment]
/lgtm
remember to upgrade the release repo and downstream dockefile.

**Pull request merged**